app/src/main/java/com/arctouch/codechallenge/home/presentation/HomeViewModelFactory.kt

```diff
@@ -0,0 +1,25 @@
+package com.arctouch.codechallenge.home.presentation
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+import com.arctouch.codechallenge.core.api.TmdbApi
+import com.arctouch.codechallenge.core.api.TmdbApiFactory
+import com.arctouch.codechallenge.home.data.repository.GenresRepository
+import com.arctouch.codechallenge.home.data.repository.MoviesRepository
+import com.arctouch.codechallenge.home.domain.HomeMoviesUseCase
+
+class HomeViewModelFactory: ViewModelProvider.Factory  {
+
+    private val api: TmdbApi by lazy {
+        TmdbApiFactory.buildTmdbApi()
+    }
+
+    private val genresRepository = GenresRepository(api)
+    private val moviesRepository = MoviesRepository(api)
+    private val moviesUseCase = HomeMoviesUseCase(genresRepository, moviesRepository)
+
+    @Suppress("UNCHECKED_CAST")
```

Comment:
This warning is there because the ViewModelProvider.Factory interface that uses a generic type for the return, i didn´t find a way to complete remove the warning without the Suppress, but researching now i discovered that i could verify using isAssignableFrom() to improve the code, something like:

@Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(HomeViewModel::class.java)) {
            return HomeViewModel(moviesUseCase) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }